src/bc/model.rs

```diff
@@ -22,10 +26,154 @@ pub enum BcBody {
     ModernMsg(ModernMsg),
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum BinaryDataKind {
+    VideoDataIframe,
+    VideoDataPframe,
+    VideoCont,
+    AudioDataAac,
+    AudioDataAdpcm,
+    AudioCont,
+    InfoData,
+    InfoDataCont,
+    Unknown,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct BinaryData {
+    pub data: Vec<u8>,
+    pub continuation_of: Option<BinaryDataKind>,
+}
+
+impl BinaryData {
+    pub fn body(&self) -> &[u8] {
+        let lower_limit = self.header_size();
+        let upper_limit = self.body_size() + lower_limit;
+        &self.data[lower_limit..upper_limit]
+    }
+
+    pub fn header_size(&self) -> usize {
```

Comment:
Oh ffs Reolink, really? A separate header size per kind?

Reply:
Yes, sorry

Reply:
Not your fault! I was angry at reolink haha.